src/app/models/field/emailField.ts

```diff
@@ -77,6 +105,15 @@ const createEmailFieldSchema = (): Schema<IEmailFieldSchema> => {
     return next()
   })
 
+  // If hasAllowedEmailDomains is false, then clear allowedEmailDomains
+  EmailFieldSchema.pre<IEmailFieldSchema>('validate', function (next) {
+    if (!this.hasAllowedEmailDomains) {
+      this.allowedEmailDomains = []
+    }
+
+    return next()
+  })
```

Comment:
how about changing the database records to be consistent with the schema, instead of performing this lazy update?

Reply:
This can happen since the frontend gives the form object wholesale. An occurrence of both `hasAllowedEmailDomains = true` and a non-empty `allowedEmailDomains` can occur when:
1. current email field has email domains set
2. admin toggles allow email domain off (frontend doesn't clear already filled email domains)
3. this scenario gets sent to the backend

Instead of changing frontend code (which may still result in inconsistencies due to client-side manipulation), I'd thought doing it here is safer and more concrete

Reply:
> frontend doesn't clear already filled email domains ... Instead of changing frontend code (which may still result in inconsistencies due to client-side manipulation), I'd thought doing it here is safer and more concrete

This reasoning isn't correct - SPAs must always maintain a consistent viewmodel and still clear `allowedEmailDomains`.

Reply:
Let me rephrase; could it be by design? A user may want to toggle it off, and toggle it back on to get their old results back before they submit? 

![recording (17)](https://user-images.githubusercontent.com/22133008/110268012-ec207100-7ffb-11eb-89e5-7feb0c93dbb6.gif)

Reply:
is this a feature or a bug? The inconsistency is also the result of the model being overly flexible, we shouldn't have introduced the `hasAllowedEmailDomains` boolean - instead, it should have only been computed on the fly by the frontend.

Reply:
in the meantime, why don't we clear the array when the toggle is touched? I don't think the UX change is going to inconvenience users.

Reply:
Will do

Reply:
FYI I created another [issue](https://github.com/opengovsg/FormSG/issues/1312) to track the inconsistency problem

Reply:
Cleared array when toggled off in d84eac1, and removed the pre-validate transformation in d665caa